Feat: Update documentation for errors and decorators (#1248)
api-platform/docs: core/errors.md

        App\Exception\ProductNotFoundException: 404 # Here is the handler for our custom exception
```

> Any type of `Exception` can be thrown, API Platform will convert it to a Symfony's `HttpException` (notice that this mean the exception will be flattened and lose all of their custom properties). The framework also takes

Reviewer: ```suggestion
Any type of `Exception` can be thrown, API Platform will convert it to a Symfony's `HttpException` (note that it means the exception will be flattened and lose all of its custom properties). The framework also takes
```
